### ISSUE TITLE
Introduction of whitespace after while and assignment operator

### ISSUE THREAD

**Issue by OJFord** [psf/black#1976]
**Describe the bug** A clear and concise description of what the bug is.

`black` produces this diff:
```diff
-            while (foo := self._some_method(arg)):
+            while (foo := self._some_method(arg)) :
```

**To Reproduce** Steps to reproduce the behavior:

1. Create a file `repro.py`:
```python
while (foo := self._some_method(arg)):
    pass
```
2. `black  repro.py ; cat repro.py`
3. Observe output:
```
reformatted repro.py
All done! ✨ 🍰 ✨
1 file reformatted.
while (foo := self._some_method(arg)) :
    pass
```

**Expected behavior**

```
All done! ✨ 🍰 ✨
1 file left unchanged.
while (foo := self._some_method(arg)):
    pass
```

**Environment (please complete the following information):**

- Version: 20.8b1
- OS and Python version: Darwin Kernel Version 20.2.0 & Python 3.9.1

**Does this bug also happen on master?** 

No.

**Additional context**

**Comment by ichard26** [comment]
This was fixed in commit 1d2d726 from PR #1655: ''Fix incorrect space before colon in if/while stmts". This issue is technically a duplicate of #1588 but I'll leave this open until a release ships with the fix mentioned above.

**Comment by ichard26** [comment]
And there has been quite a few releases since my comment above so definitely closing now. Thank you for the report!
